lib/access-context.js

```diff
@@ -88,7 +88,9 @@ function AccessContext(context) {
   var token = this.accessToken || {};
 
   if (token.userId) {
-    this.addPrincipal(Principal.USER, token.userId);
+    var userPrincipalType =
+      (this.accessToken && this.accessToken.principalType) || Principal.USER;
+    this.addPrincipal(userPrincipalType, token.userId);
```

Comment:
@akki-ng 

This change looks very reasonable to me 👍 and I am ok with landing it as an incremental improvement.

However, I need you to add a test that fails on the current master, passes with your fix in place - this will verify your fix and prevent regressions in the future.

Reply:
@bajtos Please follow on #3823